src/messages.rs

```diff
@@ -475,8 +456,30 @@ pub enum MessageContent {
         /// The message's unique identifier.
         message_id: MessageId,
     },
-    /// Send our connection_info encrypted to a node we wish to connect to and have the keys for.
-    ConnectionInfo(ConnectionInfo),
+    /// Send our Crust connection info encrypted to a node we wish to connect to and for which we
+    /// have the keys.
+    ConnectionInfoRequest {
+        /// Encrypted Crust connection info.
+        encrypted_connection_info: Vec<u8>,
+        /// Nonce used to provide a salt in the encrypted message.
+        nonce_bytes: [u8; box_::NONCEBYTES],
+        /// The sender's public ID.
+        public_id: PublicId,
+        /// The message's unique identifier.
+        message_id: MessageId,
```

Comment:
as we already generate a new `nonce` each time, will it already do the job of avoiding filtered out by using a new msg_id?